pkg/services/k8s/resources/publicdashboard/webhooks.go

```diff
@@ -152,3 +173,112 @@ func (api *WebhooksAPI) Create(c *contextmodel.ReqContext) response.Response {
 }
+
+func (api *WebhooksAPI) MutationCreate(c *contextmodel.ReqContext) response.Response {
+	api.Log.Debug("mutation controller create fired")
+	body, err := io.ReadAll(c.Req.Body)
+	if err != nil {
+		api.Log.Error("error reading request body")
+	}
+	api.Log.Debug("create", "body", string(body))
+
+	var rev k8sAdmission.AdmissionReview
+	err = json.Unmarshal(body, &rev)
+	if err != nil {
+		api.Log.Error("error unmarshalling request body")
+		return response.Error(500, "error unmarshalling request body", err)
+	}
+
+	// Current state of object
+	obj := &PublicDashboard{}
+	err = obj.UnmarshalJSON(rev.Request.Object.Raw)
+	if err != nil {
+		api.Log.Error("error unmarshalling request body")
+		return response.Error(500, "error unmarshalling request body", err)
+	}
+
+	// Previous state of object for update or delete
+	oldObj := &PublicDashboard{}
+	err = oldObj.UnmarshalJSON(rev.Request.OldObject.Raw)
+	if err != nil {
+		api.Log.Error("error unmarshalling request body")
+		return response.Error(500, "error unmarshalling request body", err)
+	}
+
+	// THIS IS BROKEN
+	// TODO: convert error to k8sAdmission.AdmissionResponse and then to response.Response
+	req := &admission.AdmissionRequest{
+		Action:  c.Req.Method,
+		Kind:    rev.Kind,
+		Group:   rev.GroupVersionKind().Group,
+		Version: rev.GroupVersionKind().Version,
+		UserInfo: admission.AdmissionUserInfo{
+			Username: rev.Request.UserInfo.Username,
+			UID:      rev.Request.UserInfo.UID,
+			Groups:   rev.Request.UserInfo.Groups,
+		},
+		Object:    obj,
+		OldObject: oldObj,
+	}
+
+	var resp *k8sAdmission.AdmissionReview
+	mutationResponse, err := api.MutationController.Mutate(c.Req.Context(), req)
+	if err != nil {
+		// auth status code to start, maybe change this to bad request
+		resp = makeFailureAdmissionReview(rev.Request.UID, rev.TypeMeta, err, 403)
+	} else {
+		resp = makeSuccessfulAdmissionReview(rev.Request.UID, rev.TypeMeta, 200)
+	}
+
+	// compare original and defaulted version
+	//ops, err := jsonpatch.CreatePatch(rev.Request.OldObject.Raw, rev.Request.Object.Raw)
+	if err != nil {
+		//responsewriters.InternalError(w, req, fmt.Errorf("unexpected diff error: %v", err))
+		//return
+	}
+	//review.Response.Patch, err = json.Marshal(ops)
+	if err != nil {
+		//responsewriters.InternalError(w, req, fmt.Errorf("unexpected patch encoding error: %v", err))
+		//return
+	}
+
+	return response.JSON(int(resp.Response.Result.Code), mutationResponse)
+}
+
+func makeSuccessfulMutationReview(uid k8sTypes.UID, typeMeta metaV1.TypeMeta, code int32) *k8sAdmission.AdmissionReview {
+
+	// patch data: json marshalled to bytes -> base64 encoded
+
+	return &k8sAdmission.AdmissionReview{
+		TypeMeta: typeMeta,
+		Response: &k8sAdmission.AdmissionResponse{
+			UID:       uid,
+			Allowed:   true,
+			PatchType: pontificate(k8sAdmission.PatchTypeJSONPatch),
+			Result: &metaV1.Status{
+				Status: "Success",
+				Code:   code,
+			},
+		},
+	}
+}
+
+// lol
+func pontificate[T any](s T) *T {
```

Comment:
I somewhat get the lol here but are we doing this for meeting a specific need instead of using the `&` directly on the site/calling loc?

Reply:
The go compiler won't let us ... soooo pontificate at least until we cleanup and refactor. This should never end up in prod

Reply:
it's already in production :lolsob: https://github.com/grafana/grafana/blob/856abe12818af590620bc967e279a8c7feece794/pkg/kindsys/util.go#L22